ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -88,29 +95,81 @@
   public static final String KSQL_COLLECT_UDF_METRICS = "ksql.udf.collect.metrics";
   public static final String KSQL_UDF_SECURITY_MANAGER_ENABLED = "ksql.udf.enable.security.manager";
 
-  private final ImmutableMap<String, Object> ksqlStreamConfigProps;
+  public static final String DEFAULT_EXT_DIR = "ext";
 
-  private static final ConfigDef CONFIG_DEF;
+  private static final Collection<CompatibiltyBreakingConfigDef> COMPATIBILTY_BREAKING_CONFIG_DEFS;
+
+  private static class CompatibiltyBreakingConfigDef {
+    private final String name;
+    private final ConfigDef.Type type;
+    private final Object defaultValueOld;
+    private final Object defaultValueCurrent;
+    private final ConfigDef.Importance importance;
+    private final String documentation;
+
+    CompatibiltyBreakingConfigDef(final String name,
+                                  final ConfigDef.Type type,
+                                  final Object defaultValueOld,
+                                  final Object defaultValueCurrent,
+                                  final ConfigDef.Importance importance,
+                                  final String documentation) {
+      this.name = name;
+      this.type = type;
+      this.defaultValueOld = defaultValueOld;
+      this.defaultValueCurrent = defaultValueCurrent;
+      this.importance = importance;
+      this.documentation = documentation;
+    }
 
-  public static final String DEFAULT_EXT_DIR = "ext";
+    public String getName() {
+      return this.name;
+    }
+
+    private void define(final ConfigDef configDef, final Object defaultValue) {
+      configDef.define(name, type, defaultValue, importance, documentation);
+    }
+
+    void defineOld(ConfigDef configDef) {
+      define(configDef, defaultValueOld);
+    }
+
+    void defineCurrent(ConfigDef configDef) {
+      define(configDef, defaultValueCurrent);
+    }
+  }
 
   static {
-    CONFIG_DEF = new ConfigDef()
+    COMPATIBILTY_BREAKING_CONFIG_DEFS = ImmutableList.of(
```

Comment:
i know it was in a static block before, but does it need to be? Can't it just be initialized at the field?